src/pages/contact.js

```diff
@@ -0,0 +1,185 @@
+/** @jsx jsx */
+import { useCallback, useState } from 'react'
+import { Redirect } from '@reach/router'
+import {
+  Button,
+  Text,
+  Box,
+  Label,
+  Input,
+  Select,
+  Textarea,
+  jsx,
+} from 'theme-ui'
+
+import Layout from '../components/layout'
+import PageHeader from '../components/pageheader'
+import Container from '../components/container'
+
+import getFirebase from '../firebase'
+
+export default function login() {
+  const firebaseApp = getFirebase()
+
+  const [error, setError] = useState(null)
+  const [success, setSuccess] = useState(null)
+
+  const [name, setName] = useState('')
+  const handleSetName = useCallback(e => setName(e.target.value), [name])
+
+  const [email, setEmail] = useState('')
+  const handleSetEmail = useCallback(e => setEmail(e.target.value), [email])
+
+  const [type, setType] = useState('Student')
+  const handleSetType = useCallback(e => setType(e.target.value), [type])
+
+  const [comment, setComment] = useState('')
+  const handleSetComment = useCallback(e => setComment(e.target.value), [
+    comment,
+  ])
+
+  const [formSubmit, setFormSubmit] = useState(false)
+
+  const submitContactForm = useCallback(
+    e => {
+      if (!firebaseApp) return
+      
+      e.preventDefault()
+
+      setError('')
+
+      if (name === '' || email === '' || type === '' || comment === '') {
+        setError('One or more fields are missing.')
+        return
+      }
+
+      if (!/\S+@\S+\.\S+/.test(email)) {
+        setError('Invalid email.')
+        return
+      }
+
+      firebaseApp
+        .firestore()
+        .collection('contact')
+        .add({
+          name: name,
+          email: email,
+          type: type,
+          comment: comment,
+        })
+        .then(() => {
+          setSuccess('Successfully submitted contact form!')
+          setTimeout(setFormSubmit.bind(null, true), 1500)
+        })
+        .catch(() => {
+          setError('Firebase error, please try again.')
+        })
+    },
+    [name, email, type, comment]
+  )
+
+  if (formSubmit) {
+    return <Redirect to='/' noThrow />
+  }
+
+  return (
+    <Layout>
+      <PageHeader primary='Contact Us' />
+      <Container>
+        <Box as='form'>
+          <Label htmlFor='name' sx={{ fontSize: [4, 5] }}>
+            Name
+          </Label>
+          <Input
+            sx={{
+              width: ['100%', null, '50%'],
```

Comment:
```suggestion
              width: '100%',
```
An input should always be the full width of its container. You can position the two inputs next to each other by wrapping an input and its label in a container, and then positioning those containers (e.g. via a grid).